src/core/server/http/http_tools.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { readFileSync } from 'fs';
+import { Server, ServerOptions } from 'hapi-latest';
+import { ServerOptions as TLSOptions } from 'https';
+import { HttpConfig } from './http_config';
+
+/**
+ * Converts Kibana `HttpConfig` into `ServerOptions` that are accepted by the Hapi server.
+ */
+export function getServerOptions(
+  config: HttpConfig,
+  { configureTLS = true } = {}
```

Comment:
Interesting. Why default to true?

Reply:
Well, it's probably arguable, but my reasoning is that `config` should dictate what server options should be produced by default and if `config` says `ssl:enabled` then corresponding options should be included, _unless_ it's specifically overriden for some reason. Does it make sense?

Reply:
Ah, yes, i see. Makes sense. So even if the config _doesn't_ have `ssl:enabled` we will try to parse them?

Reply:
> So even if the config doesn't have ssl:enabled we will try to parse the ssl config?

We parse entire config even if TLS is not enabled (https://github.com/elastic/kibana/blob/kbn-core/src/core/server/http/http_config.ts#L110).

But in this function we won't touch TLS settings if it's not enabled in the config even if `configureTLS` hint set to `true`. We just don't have use case for that. We can do that in the future if we need to, but it may be not as easy as it sounds: if TLS isn't enabled in config we may skip some validity checks during config parsing, so here we'll have to check whether it _can be_ enabled by checking that certificate and key are set an so on.

Reply:
Cool, makes sense. `configureTLS` is really like "tryToConfigureTLS"

Reply:
Yeah, exactly